src/include/miopen/norm/problem_description.hpp

```diff
@@ -0,0 +1,109 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#pragma once
+
+#include <miopen/problem_description_base.hpp>
+#include <miopen/activ.hpp>
+#include <miopen/tensor.hpp>
+
+#include <cassert>
+#include <string>
+
+namespace miopen {
+
+struct NetworkConfig;
+
+namespace norm {
+
+struct ProblemDescription : ProblemDescriptionBase
+{
+    ProblemDescription(miopenLayerNormMode_t mode_,
+                       const TensorDescriptor& xDesc_,
+                       const TensorDescriptor& weightDesc_,
+                       const TensorDescriptor& biasDesc_,
+                       const TensorDescriptor& yDesc_,
+                       const TensorDescriptor& meanDesc_,
+                       const TensorDescriptor& rstdDesc_,
+                       float epsilon_,
+                       int32_t normalized_dim_)
+        : mode(mode_),
+          xDesc(xDesc_),
+          weightDesc(weightDesc_),
+          biasDesc(biasDesc_),
+          yDesc(yDesc_),
+          meanDesc(meanDesc_),
+          rstdDesc(rstdDesc_),
+          epsilon(epsilon_),
+          normalized_dim(normalized_dim_)
+    {
+    }
+
+    miopenLayerNormMode_t GetMode() const { return mode; }
+    const TensorDescriptor& GetXDesc() const { return xDesc; }
+    const TensorDescriptor& GetWeightDesc() const { return weightDesc; }
+    const TensorDescriptor& GetBiasDesc() const { return biasDesc; }
+    const TensorDescriptor& GetYDesc() const { return yDesc; }
+    const TensorDescriptor& GetMeanDesc() const { return meanDesc; }
+    const TensorDescriptor& GetRstdDesc() const { return rstdDesc; }
+    float GetEpsilon() const { return epsilon; }
+    int32_t GetNormalizedDim() const { return normalized_dim; }
+
+    bool IsRank2Dim1() const { return (xDesc.GetLengths().size() == 2) && (normalized_dim == 1); }
+    bool IsRank4Dim1() const { return (xDesc.GetLengths().size() == 4) && (normalized_dim == 1); }
+    bool IsLargeSize() const
```

Comment:
[Recommendation] AFAICS this is not a problem attribute, but a helper function to support some solver-specific check. I highly recommend removing it from the problem.

Reply:
I misunderstood the problem description. fixed it.

Reply:
@seungmanhan 
> I highly recommend removing it from the problem.

Have you decided to keep this in the problem description? Or would like to resolve this comment in a subsequent PR?